fe/fe-core/src/main/java/com/starrocks/qe/scheduler/QueryRuntimeProfile.java

```diff
@@ -246,8 +246,13 @@ public void attachExecutionProfiles(Collection<FragmentInstanceExecState> execut
         profiles.forEach((k, v) -> fragmentProfiles.get(k).addChildren(v));
     }
 
-    public void finishInstance(TUniqueId instanceId) {
+    public synchronized void finishInstance(TUniqueId instanceId) {
```

Comment:
this might not necessary since the `markedCountDown` is already synchronized. The only usage of this `synchronized` is to ensure the RuntimeException is correctly determined.

Reply:
fixed